runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -383,4 +396,65 @@ boolean shouldFilterByName(final String playerName)
 		}
 		return false;
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.rightClickFilter())
+		{
+			return;
+		}
+
+		int type = event.getType();
+
+		if (type >= MENU_ACTION_DEPRIORITIZE_OFFSET)
+		{
+			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
+		}
+		final MenuAction menuAction = MenuAction.of(type);
+
+		if (menuAction.equals(RUNELITE_PLAYER))
```

Comment:
This will not work when no other plugin uses MenuManager to add player menu actions. (Disable the hiscore plugin to see how this doesn't work)

This plugin should also make use of [`MenuManager#addPlayerMenuItem(String menuText)`](http://static.runelite.net/api/runelite-client/net/runelite/client/menus/MenuManager.html#addPlayerMenuItem(java.lang.String)) instead of adding this entry manually.